assets/tools/asset_audit.py

```diff
@@ -59,6 +70,36 @@ def get_rpc_connection():
     rpc_connection = AuthServiceProxy(connection)
     return(rpc_connection)
 
+def log_failure(err):
+    logging.error(err)
+
+def send_notification(notification_to_emails, notification_subject, notification_content):
+    sendgrid_api_key = ''
+    if "SENDGRID_API_KEY" in os.environ:  
+      sendgrid_api_key = os.environ.get('SENDGRID_API_KEY')
+      #print("Key="+sendgrid_api_key)
+    else:
+      print("Must set SENDGRID_API_KEY environment to use e-mail notification.")
+      return
+
+    import sendgrid
+    from sendgrid.helpers.mail import Content, Email, Mail
```

Comment:
We should add all imports to the top of the file? 

Is there a reason this is in the function?

Reply:
Yes, there's a reason.  It is a conditional import.  

If you don't need notifications, and you're not adding the SENDGRID_API_KEY through the environment variable, then you shouldn't need to pip install the dependency.